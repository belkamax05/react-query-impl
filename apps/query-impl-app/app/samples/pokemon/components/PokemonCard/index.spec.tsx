import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import PokemonCard, { PokemonCardProps } from '.';
import pokemonApiSpy from '../../../../testing/spy/pokemonApiSpy';

describe('PokemonCard', () => {
  const queryClient = new QueryClient();

  const wrapper = ({ children }: { children: ReactElement }) => (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );

  it('should not fail when no data provided', async () => {
    pokemonApiSpy.useFetch.mockReturnValue({});
    const { baseElement } = render(
      <PokemonCard {...({} as PokemonCardProps)} />,
      { wrapper }
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should render the pokemon name and id', async () => {
    pokemonApiSpy.useFetch.mockReturnValue({
      data: {
        id: 25,
        name: 'pikachu',
        sprites: { front_default: 'https://img.png' },
        base_experience: 112,
      },
    });

    const pokemonName = 'pikachu';
    const pokemonId = 25;

    const { baseElement, getByText } = render(
      <PokemonCard name={pokemonName} />,
      { wrapper }
    );

    expect(getByText(`#${pokemonId} - ${pokemonName}`)).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
