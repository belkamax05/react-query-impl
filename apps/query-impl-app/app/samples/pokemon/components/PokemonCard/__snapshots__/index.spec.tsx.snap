// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard should not fail when no data provided 1`] = `
<body>
  <div>
    <div
      class="root"
    >
      <h3
        class="title"
      >
        #undefined - undefined
      </h3>
      <div
        class="img"
      >
        <img />
      </div>
      <div
        class="stats"
      >
        <pre>
          base_experience: 
        </pre>
      </div>
    </div>
  </div>
</body>
`;

exports[`PokemonCard should render the pokemon name and id 1`] = `
<body>
  <div>
    <div
      class="root"
    >
      <h3
        class="title"
      >
        #25 - pikachu
      </h3>
      <div
        class="img"
      >
        <img
          alt="pikachu"
          src="https://img.png"
        />
      </div>
      <div
        class="stats"
      >
        <pre>
          base_experience: 
          112
        </pre>
      </div>
    </div>
  </div>
</body>
`;
